src/main/java/org/example/dao/AddressDAO.java

```diff
@@ -24,15 +25,15 @@ public Address mapResultSetToObject(ResultSet resultSet) throws SQLException {
     public List<Address> getAddressesByUser(String username) {
```

Comment:
In my opinion it would be better to keep more complicated methods like this in Service layer classes.

Reply:
Due to Vladislav's suggestion, methods remained in DAO.

Reply:
Perfect!